solvation_analysis/plotting.py

```diff
@@ -0,0 +1,294 @@
+    keep_solvents : a list of strings of solvent names that are common to all systems in question,
+        graphed in the order specified by the user
+    x_label : name of x axis as a string
+    y_label : name of y axis as a string
+    title : title of figure as a string
+    legend_label : title of legend as a string
+    kwargs : consists of the x_axis and series parameters
+        x_axis : a string specifying "species" or "solution" to be graphed on the x_axis
+        series : Boolean (False for a bar graph; True for a line graph)
+
+    Returns
+    -------
+    fig : Plotly.Figure
+
+    """
+    coerce = coerce or {}
+    pairing = {solution : solutions[solution].pairing.pairing_dict for solution in solutions}
+    return compare_solvent_dicts(pairing, coerce, keep_solvents, x_label, y_label, title, legend_label, **kwargs)
+
+
+def compare_coordination_numbers(solutions, x_label, y_label, title, legend_label="Legend", coerce={}, keep_solvents=None, **kwargs):
+    # this should be a stacked bar chart, horizontal?
+    # TODO: complete docstring
+    """
+    Compares the coordination numbers of multiple solutions.
+
+    Parameters
+    ----------
+    solutions : a dictionary of Solution objects
+    x_label :
+    y_label :
+    title :
+    legend_label :
+    coerce : a dictionary where the keys are strings of solvent names and the values are
+        strings of a more generic name for the solvent (i.e. {"EAf" : "EAx", "fEAf" : "EAx"})
+    keep_solvents : a list of strings of solvent names that are common to all systems in question
+    kwargs : consists of the x_axis and series parameters
+        x_axis : a string specifying "species" or "solution" to be graphed on the x_axis
+        series : Boolean (False for a bar graph; True for a line graph)
+
+    Returns
+    -------
+    fig : Plotly.Figure
+
+    """
+    coordination = {solution: solutions[solution].coordination.cn_dict for solution in solutions}
+    return compare_solvent_dicts(coordination, coerce, keep_solvents, x_label, y_label, title, legend_label, **kwargs)
+
+
+def compare_coordination_to_random(solutions):
+    # this should compare the actual coordination numbers relative to a
+    # statistically determined "random" distribution, ignoring sterics
+    return
+
+
+def compare_residence_times(solutions, series=False, ignore=None):
+    # not in this branch yet
+    # this should be a grouped vertical bar chart or a line chart
+    """
+    Compares the coordination numbers of multiple solutions.
+
+    Parameters
+    ----------
+    solutions : a list of Solution objects
+    series : Boolean
+    ignore : list of strings of solvent names to be ignored
+
+    Returns
+    -------
+    fig : Plotly.Figure
+
+    """
+    # catch_different_solvents(solutions)
+    residence = [solution.residence.residence_times for solution in solutions]
+    residence = set(residence) - set(ignore)
```

Comment:
is this an old implementation?

Reply:
Yes, I'll update it to reflect the current implementation when I next push the code.